subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -962,6 +962,7 @@ static inline u32_t isr_rx_adv(u8_t devmatch_ok, u8_t devmatch_id,
 
 	if ((pdu_adv->type == PDU_ADV_TYPE_SCAN_REQ) &&
 	    (pdu_adv->len == sizeof(struct pdu_adv_scan_req)) &&
+	    ((pdu_adv->type != PDU_ADV_TYPE_ADV_DIRECT_IND) &&
```

Comment:
In line 963 pdu_adv->type is already checked for being SCAN_REQ, this added check is redundant. Did you want to check the Tx-ed PDU type, _pdu_adv->type?

Reply:
Yes, copy paste mistake. Thank you.

Reply:
Could refactor a bit? if it is not too much of a change to rename pdu_adv to pdu_rx, and _pdu_adv to pdu_adv. Thank you.

Reply:
There is a lot of lines, plus this is the legacy one, so I'd prefer to keep it as is.

Reply:
ok with me :-)